services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java

```diff
@@ -72,6 +72,8 @@ public class ReadRequestThrottler implements RouterThrottler, RoutersClusterMana
   private final long storageNodeQuotaCheckTimeWindow;
   private final boolean perStorageNodeThrottlerEnabled;
 
+  private boolean isNoopThrottlerEnabled;
```

Comment:
This flag can be changed in different thread, so shall we use `volatile` here?

Reply:
Sure, good safety